cmds/exp/sed/sed.go

```diff
@@ -0,0 +1,130 @@
+// Copyright 2023 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// By Ahmed Kamal <email.ahmedkamal@googlemail.com>
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path"
+	"strings"
+)
+
+type transform struct {
+	from string
+	to   string
+}
+
+type transforms []transform
+
+func (t *transforms) String() string {
+	return fmt.Sprint(*t)
+}
+
+func (t *transforms) Set(value string) error {
+	transformDefinition := strings.Split(value, "/")
+
+	if len(transformDefinition) != 3 || transformDefinition[0] != "s" {
+		return fmt.Errorf("unable to parse transformation. This should be of the form s/old/new")
+	}
+
+	*t = append(*t, transform{from: transformDefinition[1], to: transformDefinition[2]})
+
+	return nil
+}
+
+type config struct {
+	transforms transforms
+	inplace    bool
+}
+
+var cfg = config{}
```

Comment:
Please avoid global variables in command flags it makes testing hard. See /cmds/core/grep
for reference. It's better to create a structure with command flags and other necessary things. This will simplify testing significantly because it you no longer need to unwind global variables.